var/spack/repos/builtin/packages/cray-mpich/package.py

```diff
@@ -136,3 +138,83 @@ def libs(self):
         libs += find_libraries(libraries, root=self.prefix.lib64, recursive=True)
 
         return libs
+
+    # @memoized
+    @property
+    def gtl_lib(self):
+        # GPU transport Layer (GTL) handling background:
+        # - The cray-mpich module defines an environment variable per supported
+        # GPU (say, PE_MPICH_GTL_LIBS_amd_gfx942). So we should read the
+        # appropriate variable.
+        # In practice loading a module and checking its content is a PITA. We
+        # simplify by assuming that the GTL for a given vendor (say, AMD), is
+        # one and the same for all the targets of this vendor (one GTL for all
+        # Nvidia or one GTL for all AMD devices).
+        # - Second, except if you have a very weird mpich layout, the GTL are
+        # located in /opt/cray/pe/mpich/<cray_mpich_version>/gtl/lib when the
+        # MPI libraries are in
+        # /opt/cray/pe/mpich/<cray_mpich_version>/ofi/<vendor>/<vendor_version>.
+        # Example:
+        #   /opt/cray/pe/mpich/8.1.28/gtl/lib
+        #   /opt/cray/pe/mpich/8.1.28/ofi/<vendor>/<vendor_version>
+        #   /opt/cray/pe/mpich/8.1.28/ofi/<vendor>/<vendor_version>/../../../gtl/lib
+
+        gtl_kinds = [
+            [
+                "+rocm",
+                "amdgpu_target",
+                "libmpi_gtl_hsa",
+                set(["gfx906", "gfx908", "gfx90a", "gfx940", "gfx942"]),
+            ],
+            ["+cuda", "cuda_arch", "libmpi_gtl_cuda", set(["70", "80", "90"])],
```

Comment:
Can we give names to these list entries? Using subscripts here is not very meaningful, since each item in the list has a different purpose.